app/src/org/commcare/dalvik/activities/EntityMapActivity.java

```diff
@@ -1,288 +1,185 @@
-    }
-    
-    private String getLicenseKey() {
-        //If there's a defined debug key in the local environment, use that.
-        int debugId = this.getResources().getIdentifier("maps_api_key_debug","string", this.getPackageName());
-        if(debugId == 0) { 
-            return BuildConfig.MAPS_API_KEY;
-        }
-        return this.getString(debugId);
     }
 
-    private EvaluationContext entityContext;
+    private Vector<Entity<TreeReference>> getEntities(Detail detail) {
+        EvaluationContext evaluationContext = session.getEvaluationContext(
+                new AndroidInstanceInitializer(session));
+        evaluationContext.addFunctionHandler(EntitySelectActivity.getHereFunctionHandler());
+
+        NodeEntityFactory factory = new NodeEntityFactory(detail, evaluationContext);
+        Vector<TreeReference> references = evaluationContext.expandReference(
+                selectDatum.getNodeset());
 
-    private EvaluationContext getEC() {
-        if(entityContext == null) {
-            entityContext = session.getEvaluationContext(getInstanceInit());
+        Vector<Entity<TreeReference>> entities = new Vector<>();
+        for(TreeReference ref : references) {
+            entities.add(factory.getEntity(ref));
         }
-        return entityContext;
+        return entities;
     }
-    
-    private AndroidInstanceInitializer getInstanceInit() {
-        return new AndroidInstanceInitializer(session);
+
+    private LatLng getLatLngFromAddress(@NonNull String address) {
+        LatLng location = null;
+        try {
+            GeoPointData data = new GeoPointData().cast(new UncastData(address));
+            if(data != null) {
+                location = new LatLng(data.getLatitude(), data.getLongitude());
+            }
+        } catch(IllegalArgumentException ignored) {
+        }
+        return location;
     }
 
     @Override
-    protected void onStop() {
-        super.onStop();
-        mMyLocationOverlay.disableCompass(); 
-        mMyLocationOverlay.disableMyLocation();
+    public void onMapReady(final GoogleMap map) {
+        mMap = map;
+
+        if (entityLocations.size() > 0) {
+            LatLngBounds.Builder builder = new LatLngBounds.Builder();
+            // Add markers to map and find bounding region
+            for (Pair<Entity<TreeReference>, LatLng> entityLocation : entityLocations) {
+                Marker marker = mMap.addMarker(new MarkerOptions()
+                        .position(entityLocation.second)
+                        .title(entityLocation.first.getFieldString(0))
+                        .snippet(entityLocation.first.getFieldString(1)));
+                markerReferences.put(marker, entityLocation.first.getElement());
+                builder.include(entityLocation.second);
+            }
+            final LatLngBounds bounds = builder.build();
+
+            // Move camera to be include all markers
+            // TODO: does this work for 1 marker?

```

Comment:
Can you check this and remove it before we merge? Should be able to check by adding a `return` below https://github.com/dimagi/commcare-odk/blob/update-map-api/app/src/org/commcare/dalvik/activities/EntityMapActivity.java#L84